Assets/Scripts/Player/PlayerMovement.cs

```diff
@@ -15,34 +15,72 @@ public class PlayerMovement : MonoBehaviour
     private Vector3 movement;
     private CharacterController controller;
 
+    private Camera cam;
+
     private void Awake()
     {
         controller = GetComponent<CharacterController>();
+
+        // TC: cache the camera reference
+        cam = Camera.main;
     }
 
     //read movement inputs
     private void OnMove(InputValue value)
     {
-        movement.x = value.Get<Vector2>().x;
-        movement.z = value.Get<Vector2>().y;
-        movement = movement.normalized;
+        // TC: store the movement vector in one single get and normalize call
+        movement = value.Get<Vector2>().normalized;
+
+        //movement.x = value.Get<Vector2>().x;
+        //movement.z = value.Get<Vector2>().y;
+        //movement = movement.normalized;
     }
 
     //move
     private void Update()
     {
-        //Apply movement
-        AlignMovement(movement);
-        controller.Move(new Vector3(movement.x, 0f, movement.z) * moveSpeed * Time.deltaTime);
+        var finalMovement = CalculateMovementVector();
+        controller.Move(finalMovement);
+        RotatePlayerGFX(finalMovement);
+
+
+        ////Apply movement
+        //AlignMovement(movement);
+        //controller.Move(new Vector3(movement.x, 0f, movement.z) * moveSpeed * Time.deltaTime);
+
+        //if (movement != Vector3.zero && playerModel != null)
+        //{
+        //    //Rotate the player model towards the movement direction
+        //    Quaternion targetRotation = Quaternion.LookRotation(new Vector3(movement.x, 0, movement.z), Vector3.up);
+        //    playerModel.transform.rotation = Quaternion.Lerp(playerModel.transform.rotation, targetRotation, Time.deltaTime * rotationSpeed);
+        //}
+    }
 
-        if(movement != Vector3.zero && playerModel != null)
+    private void RotatePlayerGFX(Vector3 finalMovement)
+    {
+        // TC: Rotate the player model towards the movement direction
+        if (finalMovement != Vector3.zero)
         {
-            //Rotate the player model towards the movement direction
-            Quaternion targetRotation = Quaternion.LookRotation(new Vector3(movement.x, 0, movement.z), Vector3.up);
-            playerModel.transform.rotation = Quaternion.Lerp(playerModel.transform.rotation, targetRotation, Time.deltaTime * rotationSpeed);
+            Quaternion targetRotation = Quaternion.LookRotation(finalMovement);
+            targetRotation.x = 0;
+            targetRotation.z = 0;
```

Comment:
determine the where the playerGFX should rotate to using `LookRotation` at the `finalMovement` and then zero out the vector values to ensure it only rotates in the Y axis